### DISCUSSION TITLE
Build error using VSCode on leap 15.6

### DISCUSSION THREAD

**Discussion by tzdisco4** [emsesp/EMS-ESP32#2110] [Q&A] [answered]
Hi all,
I tried to build emsesp from a git clone using vscode with platformio an got this error:
`#include "WWWData.h" // include auto-generated static web resources
          ^~~~~~~~~~~
compilation terminated.`
I found an old issue from 2020 related to node.js mentioning that the missing file would be build by npm automatically initiated by platformio.ini. But i couldn't find any hints in the ini file regarding npm.

I installed nvm, nmp and node, but it seams that it isn't used during build process.

Any Ideas?
Thanks

**Comment by proddy** [comment, marked as answer]
`Pio run' will, by default, use the GitHub actions CI targets. If you're building locally, you need to choose the right target for PIO: either `s3_16M_P` if it's an ESP32-S3 (for a BBQKees S3) or `s_16M_P` for an ESP32 (like the BBQKees E32V2).

See https://emsesp.org/Building/

@MichaelDvP shall we just change the default envs for pio to build all standalone targets and not the CI/GH Action ones?

**Reply by tzdisco4** [reply]
@proddy Thanks for your fast answer! After learning a bit more about pio and understanding your answer I could overcome a few further hurdles and finally build s_16M_P :).
